tutorials/colors/colormap-manipulation.py

```diff
@@ -0,0 +1,196 @@
+
+##############################################################################
+# The colormap is a lookup table, so "oversampling" the colormap returns
+# nearest-neighbor interpolation (note the repeated colors in the list below)
+
+print('viridis(np.linspace(0, 1, 15))', viridis(np.linspace(0, 1, 15)))
+
+##############################################################################
+# Creating a new ListedColormap
+# =============================
+#
+# This is essential the inverse operation of the above where we supply a
+# Nx4 numpy array with all values between 0 and 1,
+# to `.ListedColormap` to make a new colormap.  This means that
+# any numpy operations that we can do on a Nx4 array make carpentry of
+# new colormaps from existing colormaps quite straight forward.
+#
+# Suppose we want to make the first 25 entries of a 256-length "viridis"
+# colormap pink for some reason:
+
+viridis = cm.get_cmap('viridis', 256)
+newcolors = viridis(np.linspace(0, 1, 256))
+pink = np.array([248/256, 24/256, 148/256, 1])
+newcolors[:25, :] = pink
+newcmp = ListedColormap(newcolors)
+
+
+def plot_examples(cms):
+    """
+    helper function to plot two colormaps
+    """
+    np.random.seed(19680801)
+    data = np.random.randn(30, 30)
+
+    fig, axs = plt.subplots(1, 2, figsize=(6, 3), constrained_layout=True)
+    for [ax, cmap] in zip(axs, cms):
+        psm = ax.pcolormesh(data, cmap=cmap, rasterized=True, vmin=-4, vmax=4)
+        fig.colorbar(psm, ax=ax)
+    plt.show()
+
+plot_examples([viridis, newcmp])
+
+##############################################################################
+# We can easily reduce the dynamic range of a colormap; here we choose the
+# middle 0.5 of the colormap.  However, we need to interpolate from a larger
+# colormap, otherwise the new colormap will have repeated values.
+
+viridisBig = cm.get_cmap('viridis', 512)
+newcmp = ListedColormap(viridisBig(np.linspace(0.25, 0.75, 256)))
+plot_examples([viridis, newcmp])
+
+##############################################################################
+# and we can easily concatenate two colormaps:
+
+top = cm.get_cmap('Oranges_r', 128)
+bottom = cm.get_cmap('Blues', 128)
+
+newcolors = np.vstack((top(np.linspace(0, 1, 128)),
+                       bottom(np.linspace(0, 1, 128))))
+newcmp = ListedColormap(newcolors, name='OrangeBlue')
+plot_examples([viridis, newcmp])
+
+##############################################################################
+# Of course we need not start from a named colormap, we just need to create
+# the Nx4 array to pass to `.ListedColormap`.  Here we  create a
+# brown colormap that goes to white....
+
+N = 256
+vals = np.ones((N, 4))
+vals[:, 0] = np.linspace(90/256, 1, N)
+vals[:, 1] = np.linspace(39/256, 1, N)
+vals[:, 2] = np.linspace(41/256, 1, N)
+newcmp = ListedColormap(vals)
+plot_examples([viridis, newcmp])
+
+##############################################################################
+# LinearSegmented colormaps
```

Comment:
1. Would call this "Creating a new LinearSegmentedColormap" in resemblence to "Creating a new ListedColormap".

2. Not sure if we need the "new" in the title.